webapp/app/forms/fields.py

```diff
@@ -19,6 +19,44 @@ def pre_validate(self, form):
         ValidElsterCharacterSet().__call__(form, self)
 
 
+class UnlockCodeWidget(TextInput):
+    """A divided input field with three text input fields, limited to four chars."""
+
+    def __call__(self, field, **kwargs):
+        if 'required' not in kwargs and 'required' in getattr(field, 'flags', []):
+            kwargs['required'] = True
+
+        kwargs['maxlength'] = 4
+        kwargs['class'] = 'form-control'
+
+        html = ""
+        kwargs['value'] = field._value()[0]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
+        html += '-'
+        kwargs['value'] = field._value()[1]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
+        html += '-'
+        kwargs['value'] = field._value()[2]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
```

Comment:
Couldn't we set these in a for loop and set a class attribute `number_of_fields` and then this would that also make this more adaptable?

Reply:
see below